src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    /// <summary>
+    /// CA1862: Prefer the StringComparison method overloads to perform case-insensitive string comparisons.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class RecommendCaseInsensitiveStringComparisonAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1862";
+
+        internal const string StringComparisonInvariantCultureIgnoreCaseName = "InvariantCultureIgnoreCase";
+        internal const string StringComparisonCurrentCultureIgnoreCaseName = "CurrentCultureIgnoreCase";
+        internal const string StringToLowerMethodName = "ToLower";
+        internal const string StringToUpperMethodName = "ToUpper";
+        internal const string StringToLowerInvariantMethodName = "ToLowerInvariant";
+        internal const string StringToUpperInvariantMethodName = "ToUpperInvariant";
+        internal const string StringContainsMethodName = "Contains";
+        internal const string StringIndexOfMethodName = "IndexOf";
+        internal const string StringStartsWithMethodName = "StartsWith";
+        internal const string StringCompareToMethodName = "CompareTo";
+
+        private static readonly LocalizableString s_localizableStringComparisonTitle = CreateLocalizableResourceString(nameof(RecommendCaseInsensitiveStringComparisonTitle));
```

Comment:
I think we inline localizable strings when they are used once only. See https://github.com/dotnet/roslyn-analyzers/pull/5406